### ISSUE TITLE
cucumber with javascript?

### ISSUE THREAD

**Issue by cnishina** [angular/protractor-cookbook#18]
@igniteram I defer to your expertise if you want to provide an example.  Also there's a [SO question](http://stackoverflow.com/questions/41136296/e-launcher-error-typeerror-cucumber-cli-is-not-a-function) with Protractor + cucumber 2.0.0-rc2. Do you know anything about cucumber 2.0.0?

**Comment by igniteram** [comment]
Yes, in cucumber 2.0.0 some breaking changes were introduced to the cucumber cli which could be found in their [changelog.md](https://github.com/cucumber/cucumber-js/blob/master/CHANGELOG.md#200-rc0-2016-11-25). Now this is breaking protractor-cucumber-framework, an issue has already been logged [issue 64](https://github.com/mattfritz/protractor-cucumber-framework/issues/64).

**Comment by igniteram** [comment]
A quick update on this- There is a PR for this here - [66](https://github.com/mattfritz/protractor-cucumber-framework/pull/66), its still in review. soon cucumber 2.0 support should be available!

**Comment by westlakem** [comment]
I would like to point out how disappointed I am that jasmine is just now getting cucumber 2 support.  It's been out with an official release since March 2015 for Ruby.  Almost 2 years later and still no Cucucmber 2 support.

**Comment by cnishina** [comment]
The protractor cucumber framework project appears to be open source. You could help by contributing. Disappointment does not make a project go faster.

**Comment by igniteram** [comment]
@westlakem Cucumber project doesn't follow same release cycles for various language implementations, it was originally developed with ruby so it has more support for it than javascript. We are currently evaluating all scenarios to support cucumber 2.0 with protractor as lot of breaking changes were introduced.

As pointed out by @cnishina since this is an open source project, it would take some time. You could also help us to bring this up faster :+1:

**Comment by igniteram** [comment]
An update here-  Since the author of the [protractor-cucumber-framework](https://github.com/mattfritz/protractor-cucumber-framework) is not responding on the PR's, A fork with cucumber 2 support has been created and could downloaded be from [here](https://www.npmjs.com/package/%40darrinholst%2Fprotractor-cucumber-framework), it is backward compatible and supports older & newer cucumber versions. Meanwhile I will be sending the examples on how to use protractor with cucumber2.0. So this cookbook is up to date.
